documentation/course/e2/README.md

```diff
@@ -26,53 +28,738 @@ Kluczowe ograniczenia i decyzje projektowe zawarte w dokumencie wynikają z potr
+| `NF/REL/02`, `NF/REL/07`, `NF/REL/09`, `M/04`              | [`M/05`: Kolejki SQS dla płatności i emaili](#m05-kolejki-sqs-dla-płatności-i-emaili) |
+| `NF/REL/04`, `NF/REL/06`, `M/01`, `M/04`                   | [`M/06`: Izolacja siecią wewnętrzną VPC](#m06-izolacja-siecią-wewnętrzną-vpc)         |
+| `NF/REL/04`, `M/01`, `M/06`                                | [`M/07`: Wzorzec API Gateway](#m07-wzorzec-api-gateway)                               |
+| `PAY/11`, `PAY/12`, `PAY/14`, `PAY/16`, `NF/REL/09`        | [`M/08`: Zewnętrzna bramka płatności](#m08-zewnętrzna-bramka-płatności)               |
+| `NF/REL/01`, `NF/REL/07`, `M/01`                           | [`M/09`: Oddzielne bazy dla mikroserwisów](#m09-oddzielne-bazy-dla-mikroserwisów)     |
+| `NF/REL/01`, `NF/REL/02`, `NF/REL/03`, `NF/REL/07`, `M/04` | [`M/10`: Relacyjne bazy danych ACID na RDS](#m10-relacyjne-bazy-danych-acid-na-rds)   |
+| `ACC/01`, `ACC/02`, `ACC/03`, `NF/REL/04`                  | [`M/11`: Autoryzacja z użyciem JWT](#m11-autoryzacja-z-użyciem-jwt)                   |
+| `NF/PRF/01`, `NF/PRF/02`, `M/11`                           | [`M/12`: Wzorzec sidecar dla autoryzacji](#m12-wzorzec-sidecar-dla-autoryzacji)       |
+| `NF/SYS/01`, `NF/SYS/02`, `NF/SYS/03`                      | [`M/13`: Responsywna aplikacja webowa SPA](#m13-responsywna-aplikacja-webowa-spa)     |
+| `TIC/09`, `TIC/12`, `M/13`                                 | [`M/14`: Kod QR dla biletów](#m14-kod-qr-dla-biletów)                                 |
 
-# Decyzje i ich uzasadnienie
+# Mechanizmy architektoniczne
 
-<!--
-[List the decisions that have been made regarding architectural approaches and the constraints being placed on the way that the developers build the system. These will serve as guidelines for defining architecturally significant parts of the system. Justify each decision or constraint so that developers understand the importance of building the system according to the context created by those decisions and constraints. This may include a list of DOs and DON’Ts to guide the developers in building the system.]
+## `M/01`: Architektura mikroserwisów
 
-Oba przykłady wskazują po kilka możliwych rozwiązań dla każdego celu.
--->
+TODO @tchojnacki: porównanie z monolitem i modularnym monolitem
 
-TODO @everyone
+**Problem:**
 
-| **Cel** | **Sposób osiągnięcia (taktyki)** |
-| ------- | -------------------------------- |
-|         |                                  |
+**Rozwiązania:**
 
-# Mechanizmy architektoniczne
+<table>
+  <tr>
+    <td></td>
+    <th>Zalety</th>
+    <th>Wady</th>
+  </tr>
+  <tr>
+    <th>Rozwiązanie 1</th>
+    <td>
+      <ul>
+        <li>Zaleta 1</li>
+      </ul>
+    </td>
+    <td>
+      <ul>
+        <li>Wada 1</li>
+      </ul>
+    </td>
+  </tr>
+  <tr>
+    <th>Rozwiązanie 2</th>
+    <td>
+      <ul>
+        <li>Zaleta 1</li>
+      </ul>
+    </td>
+    <td>
+      <ul>
+        <li>Wada 1</li>
+      </ul>
+    </td>
+  </tr>
+</table>
```

Comment:
For trivial decisions with no real alternatives (e.g. healthchecks), we can omit the table...